storage/files.rb

```diff
@@ -65,6 +77,29 @@ def upload_file project_id:, bucket_name:, local_file_path:,
   # [END upload_file]
 end
 
+def upload_encrypted_file project_id:, bucket_name:, local_file_path:,
+                          storage_file_path: nil, base64_encryption_key:
+  # [START upload_encryption_file]
+  # project_id            = "Your Google Cloud project ID"
+  # bucket_name           = "Your Google Cloud Storage bucket name"
+  # local_file_path       = "Path to local file to upload"
+  # storage_file_path     = "Path to store the file in Google Cloud Storage"
+  # base64_encryption_key = "Base64 encoded AES-256 encryption key"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+
+  bucket = storage.bucket bucket_name
+
+  encryption_key = Base64.decode64 base64_encryption_key
```

Comment:
Handle user input/output in the "main() method" where we call the method, not inside the snippet, itself.

For example, if we want to have a code sample that has an integer as a variable, I might...

```ruby
def add_numbers a:, b:
  # [START add_numbers]
  # a = 1  # Number to add
  # b = 2  # Number to add

  result = a + b

  puts "Result: #{result}"
  # [END add_numbers]
end

if __FILE__ == $PROGRAM_NAME
  command = ARGV.shift

  case command
  when "add"
    first_number  = ARGV.shift.to_i   # <---- convert user input here  😎
    second_number = ARGV.shift.to_i

    add_numbers a: first_number, b: second_number
  else
    puts "Unknown command: #{command}"
  end
end
```

----

**P.S.** Another way to think about it... In a CLI application, think of the main method as a "controller."  It handles user input/output.  Then it operates on models.  In our case, these are our snippets.

**Goal:** prioritize the experience of *reading* the snippets on cloud.google.com so they make the most sense.  The CLI application is secondary.